### PR TITLE
Use file interface instead of requiring os.File

### PR DESCRIPTION
**Pull request by hinshun**
I have some downstreams that won't be able to provide `os.File` but something close to it.

### CODE REVIEW

**Comment by muesli** [comment]
Probably makes sense. Curious, what types / implementations of this interface are you referring to?

File: README.md [hunk 1]
```diff
@@ -14,9 +14,9 @@ in Charm's [Bubble Tea](https://github.com/charmbracelet/bubbletea) framework.
 ## Usage
 
 `NewReader` returns a reader with a `Cancel` function. If the input reader is an
-`*os.File`, the cancel function can be used to interrupt a blocking `Read` call.
+`File`, the cancel function can be used to interrupt a blocking `Read` call.
 In this case, the cancel function returns true if the call was canceled
-successfully. If the input reader is not an `*os.File`, the cancel function does
+successfully. If the input reader is not an `File`, the cancel function does
```
Comment (muesli) [inline README.md:19]: This is probably a problem throughout these changes: `an File` -> `a File`.

**Review by muesli — CHANGES_REQUESTED** [review]

**Comment by hinshun** [comment]
For example containerd's `console.Console` or `console.File`:
https://github.com/containerd/console/blob/main/console.go#L36

File: cancelreader_windows.go [hunk 2]
```diff
@@ -17,14 +17,14 @@ import (
 var fileShareValidFlags uint32 = 0x00000007
 
 // NewReader returns a reader and a cancel function. If the input reader is an
```
Comment (muesli) [inline cancelreader_windows.go:19]: One more "an" here.

File: README.md [hunk 3]
```diff
@@ -14,9 +14,9 @@ in Charm's [Bubble Tea](https://github.com/charmbracelet/bubbletea) framework.
 ## Usage
 
 `NewReader` returns a reader with a `Cancel` function. If the input reader is an
```
Comment (muesli) [inline README.md:16]: One more "an" here.

**Review by muesli — CHANGES_REQUESTED** [review]

File: cancelreader_bsd.go [hunk 4]
```diff
@@ -14,13 +14,13 @@ import (
 )
 
 // NewReader returns a reader and a cancel function. If the input reader is an
```
Comment (muesli) [inline cancelreader_bsd.go:16]: One more `an` here. Apologies! :smile:

File: cancelreader_linux.go [hunk 5]
```diff
@@ -14,13 +14,13 @@ import (
 )
 
 // NewReader returns a reader and a cancel function. If the input reader is an
```
Comment (muesli) [inline cancelreader_linux.go:16]: One more "an" here.

**Pull request merged**